### ISSUE TITLE
Cannot run on mac

### ISSUE THREAD

**Issue by CitronDevo** [lukaszliniewicz/Pandrator#8]
While trying your app, when  I'm selecting a text, here's the error message
```
python pandrator.py
/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
pygame 2.5.2 (SDL 2.28.3, Python 3.12.3)
Hello from the pygame community. https://www.pygame.org/contribute.html
This plataform is not currently supported, check 'support.txt' file for
            information on how to run this class on other platforms.
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000018448e800 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x0000000183f85eb4 objc_exception_throw + 60
	2   CoreFoundation                      0x00000001843ba834 -[__NSArrayM insertObject:atIndex:] + 1284
	3   libtk8.6.dylib                      0x000000010323b6d0 setAllowedFileTypes + 268
	4   libtk8.6.dylib                      0x000000010323c190 Tk_GetOpenFileObjCmd + 1316
	5   libtcl8.6.dylib                     0x00000001034a7bec Tcl_EvalObjv + 340
	6   _tkinter.cpython-312-darwin.so      0x0000000102f21c5c Tkapp_Call + 320
	7   Python                              0x0000000104052a30 cfunction_call + 108
	8   Python                              0x0000000103fe4d64 _PyObject_Call + 164
	9   Python                              0x000000010412ad24 _PyEval_EvalFrameDefault + 54472
	10  Python                              0x0000000103fe7e70 method_vectorcall + 180
	11  Python                              0x000000010412ad24 _PyEval_EvalFrameDefault + 54472
	12  Python                              0x0000000103fe7e70 method_vectorcall + 180
	13  _tkinter.cpython-312-darwin.so      0x0000000102f25284 PythonCmd + 196
	14  libtcl8.6.dylib                     0x00000001034a7bec Tcl_EvalObjv + 340
	15  libtcl8.6.dylib                     0x00000001034a9128 TclEvalEx + 2192
	16  libtk8.6.dylib                      0x000000010317ef98 Tk_BindEvent + 5808
	17  libtk8.6.dylib                      0x0000000103185990 TkBindEventProc + 336
	18  libtk8.6.dylib                      0x000000010318d6a4 Tk_HandleEvent + 832
	19  libtk8.6.dylib                      0x000000010318dc70 WindowEventProc + 112
	20  libtcl8.6.dylib                     0x000000010356032c Tcl_ServiceEvent + 180
	21  libtcl8.6.dylib                     0x0000000103560594 Tcl_DoOneEvent + 248
	22  _tkinter.cpython-312-darwin.so      0x0000000102f25c60 _tkinter_tkapp_mainloop_impl + 260
	23  Python                              0x0000000103ff4eb4 method_vectorcall_FASTCALL + 116
	24  Python                              0x0000000104128e9c _PyEval_EvalFrameDefault + 46656
	25  Python                              0x0000000103fe7f2c method_vectorcall + 368
	26  Python                              0x000000010412ad24 _PyEval_EvalFrameDefault + 54472
	27  Python                              0x000000010411d4c0 PyEval_EvalCode + 304
	28  Python                              0x000000010419d6ec run_mod + 176
	29  Python                              0x000000010419c77c pyrun_file + 148
	30  Python                              0x000000010419b6e0 _PyRun_SimpleFileObject + 304
	31  Python                              0x000000010419b350 _PyRun_AnyFileObject + 248
	32  Python                              0x00000001041c9878 pymain_run_file_obj + 220
	33  Python                              0x00000001041c95b8 pymain_run_file + 72
	34  Python                              0x00000001041c8cfc Py_RunMain + 1396
	35  Python                              0x00000001041c8e98 pymain_main + 40
	36  Python                              0x00000001041c9018 Py_BytesMain + 40
	37  dyld                                0x0000000183fc10e0 start + 2360
)
libc++abi: terminating due to uncaught exception of type NSException
[1]    2532 abort      /Library/Frameworks/Python.framework/Versions/3.12/bin/python3 pandrator.py
```
Does it mean that this app does not support mac ? Thank you

**Comment by lukaszliniewicz** [comment]
Please try running it with Python 3.11. If you don't know how to do that, Claude Sonnet (the free model) should be able to guide you through the steps. Please let me know if it works.

**Comment by lukaszliniewicz** [comment]
Unfortunately, I couldn't find anything. A similar problem was mentioned, but should have been patched in Python 3.12.1, so you shouldn't be affected by it. Perhaps try running it in a Linux or Windows virtual machine and the API server(s) on your mac host?

**Comment by CitronDevo** [comment]
Thank you for your answers. I will give it a shot.
Looks like a great app, congrats!
